clients/src/main/java/org/apache/kafka/common/KafkaFuture.java

```diff
@@ -35,18 +35,26 @@
     /**
      * A function which takes objects of type A and returns objects of type B.
      */
-    public static abstract class Function<A, B> {
-        public abstract B apply(A a);
+    public interface FunctionInterface<A, B> {
+        B apply(A a);
     }
+    
+    /**
+     * A function which takes objects of type A and returns objects of type B.
+     *
+     * Replaced by {@link FunctionInterface}.
```

Comment:
You need an `@deprecated` javadoc tag here as well as the annotation.

Reply:
Missed it, update coming.